xonsh/lexer.py

```diff
@@ -30,7 +30,7 @@
         # basic operators
         '+': 'PLUS', '-': 'MINUS', '*': 'TIMES', '/': 'DIVIDE',
         '//': 'DOUBLEDIV', '%': 'MOD', '**': 'POW', '|': 'PIPE',
-        '&': 'AMPERSAND', '~': 'TILDE', '^': 'XOR', '<<': 'LSHIFT',
+        '~': 'TILDE', '^': 'XOR', '<<': 'LSHIFT',
```

Comment:
It's not actually gone; it's just being handled in a different way because it can now have some special meanings beyond just ``AMPERSAND`` (that is, it might be a ``IOREDIRECT``).

The ``handle_ampersand`` function is what is actually responsible for handling it, and I believe that is working as expected (``&`` in Python mode still works).

```
>>> 7 & 2
2
>>> 7 &> 2
xonsh: subprocess mode: command not found: 7
```